### ISSUE TITLE
Bug: Duplicate `extends` clause for `interface` should be reported

### ISSUE THREAD

**Issue by leaysgur** [typescript-eslint/typescript-eslint#11077]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&fileType=.tsx&code=MYGwhgzhAECiCM0DeBfAUKSNYCZno3CmgA1EBTADwBdyA7AE20VTQHo3pNiS8rbGzaP3pM4eVtxgkAzNACWAWwAOIcovrUhrDlyLSALApVqNdLXERLV6zdgnoppAKzCaooddN3x%2BdpycSADZjGzMLBDcBMVw-NHlzcgAnADMwYHJoAE0Kd0FLPwTaVPTMrL48mNzo%2B3xoXUBeDcABHegAFQALeRgIdoB7AFcQBmgAI0yk8mVepNphkYBPaGUwJIhkgBpoEHkAa0ze6nbkiH8FRJKM7LkvW3NtdF0i5LTLrKMb8M8TW4iHU6eLmVXB8fJERPlYjpOACXmUQuCqqFvHdfKggA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class E1 {}
class E2 {}

// class X2 extends E1 extends E2 {} // This is reported
interface Y2 extends E1 extends E2 {} // This is NOT reported
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Parser should report internal error `TS1172: 'extends' clause already seen.` like TS does.

https://www.typescriptlang.org/play/?#code/MYGwhgzhAECiCM0DeBfAUKSNYCZno3CmgA1EBTADwBdyA7AE20VTQHo3pNiS8rbGzaP3pM4eVtxgkAzNACWAWwAOIcovrUhrDlyLSALApVqNdLXERLV6zdgnoppAKzCaooddN3x+dpycSADZjGzMLBDcBMVw-NHlzcgAnADMwYHJoAE0Kd0FLPwTaVPTMrL48mNzo+z9dIuS0jOy5L1tzbXR6xJLmrKM28M8TdoiHfwUeprLXQZ9IkXzYnU4G3rKQxarQ7w7fVCA

### Actual Result

No errors are reported.

### Additional Info

Since duplicate `extends` are reported correctly for `class`, they should be reported for `interface` too?

https://github.com/typescript-eslint/typescript-eslint/blob/7f327630988f35a52752936a4fa8aa602de8e04b/packages/typescript-estree/src/convert.ts#L2043-L2049

I can send a PR if this is acceptable change.

https://github.com/leaysgur/typescript-eslint/commit/0a6d4db24cee1037a8826ec4e9304820d1533d68

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.30.1` |

**Comment by bradzacher** [comment]
I would direct you towards my reply on your last issue https://github.com/typescript-eslint/typescript-eslint/issues/11064#issuecomment-2801687992

TS allows a myriad of combinations of invalid syntax - most of which we don't explicitly fail on either. Linting code with any TS errors is considered "undefined behaviour".

The reason we have errors for classes is because there's some syntactical ordering to classes that should hold true - eg `implements` comes after `extends` - and us implementing the checks was easier than downstream consumers like prettier implementing the checks. And considering we were adding a few checks adding more was easy and made sense.

It's really a question of what is to be gained from adding the errors. Why do you think this should be an explicit and hard crash?

**Comment by leaysgur** [comment]
Thank you for your continued kindness. 🙏

> Why do you think this should be an explicit and hard crash?

This was motivated by:

- For `interface`
- duplicate `implements` are reported as internal error
- But, why duplicate `extends` are not?

I was simply concerned about these difference.

**Comment by JoshuaKGoldberg** [comment]
Agreed. As with #11088, 👍 from me on adding slightly more sanitization to the AST.

**Comment by fisker** [comment]
Fixed in [#11271](https://github.com/typescript-eslint/typescript-eslint/pull/11271)?
